lib/cloud_controller/metrics/periodic_updater.rb

```diff
@@ -84,13 +85,14 @@ def update_job_queue_length
       end
 
       pending_job_count_by_queue.reverse_merge!(@known_job_queues)
-      @updaters.each { |u| u.update_job_queue_length(pending_job_count_by_queue, total) }
+      @statsd_updater.update_job_queue_length(pending_job_count_by_queue, total)
+      @prometheus_updater.update_job_queue_length(pending_job_count_by_queue)
     end
 
     def update_thread_info
       local_thread_info = thread_info
 
-      @updaters.each { |u| u.update_thread_info(local_thread_info) }
+      [@statsd_updater, @prometheus_updater].each { |u| u.update_thread_info(local_thread_info) }
```

Comment:
I'm not sure if we should add this "thread info" thing to the prometheus world. This is very Thin/EventMachine specific and our goal should be to make the `prometheus_updater` compatible with Puma right from the start.